cg/meta/clean/clean_retrieved_spring_files.py

```diff
@@ -0,0 +1,39 @@
+import datetime
+import logging
+from pathlib import Path
+
+from housekeeper.store.models import File
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+
+LOG = logging.getLogger(__name__)
+
+
+class CleanRetrievedSpringFilesAPI:
+    """API for cleaning archived Spring files which have been retrieved."""
+
+    def __init__(self, housekeeper_api: HousekeeperAPI):
+        self.housekeeper_api: HousekeeperAPI = housekeeper_api
+
+    def _get_files_to_remove(self) -> list[File]:
+        """Returns all Spring files which were retrieved more than 7 days ago."""
+        return self.housekeeper_api.get_spring_files_retrieved_before(
+            date=datetime.datetime.now() - datetime.timedelta(days=7)
+        )
```

Comment:
should this be the age of files on server or retrieval date? What happens when the retrieval did not work as intended (i.e. the API was down for maintenance or so).

Reply:
Currently the way it works with retrievals is as follows:
1. Someone runs a start command which sees that some needed spring files are archived.
2. A retrieval job is submitted and an error is raised for the start command, stating that files are missing. At this point only the `retrieval_task_id` is set.
3. At regular intervals we query ongoing archivals and retrievals. Given that we get a "Completed" response via Miria, we set `retrieved_at` (or `archived_at` for archivals). The current implementation does not check at what time the file was added to Hasta, so the timestamp will be whenever the query has been run and given a successful response. I do not see this discrepancy mattering, given that we intend to query the ongoing jobs at quite frequent intervals.

So we might have a small discrepancy between the time when the file was added to Hasta and when the `retrieved_at` was set, but not enough to matter in my opinion. And we only set retrieved_at when we have gotten a confirmation via Miria that the job has finished.